lib/internal/http2/core.js

```diff
@@ -3098,7 +3098,7 @@ function initializeTLSOptions(options, servername) {
   options.ALPNProtocols = ['h2'];
   if (options.allowHTTP1 === true)
     ArrayPrototypePush(options.ALPNProtocols, 'http/1.1');
-  if (servername !== undefined && options.servername === undefined)
+  if (servername !== undefined && !options.servername)
```

Comment:
I don't know it makes sense for http2 but for https [the empty string is used to disable the SNI extension](https://github.com/nodejs/node/commit/98e9de7db930e505884ca8d0ca6588afbe43f127). I wonder if the same should be done here.

@nodejs/http2

Reply:
Thanks. Yeah that makes sense. This PR is based in what the issue poster has expected to be the intended behaviour. If you confirm SNI should be disabled, I would be happy to make these changes.